obj/gebaeude.cc

```diff
@@ -403,7 +404,13 @@ image_id gebaeude_t::get_image() const
 
 image_id gebaeude_t::get_outline_image() const
 {
-	if(env_t::hide_buildings!=0  &&  env_t::hide_with_transparency  &&  !zeige_baugrube) {
+	// get city highlight & city highlighted
+	city_info_t* w_city = dynamic_cast<city_info_t*>(win_get_top());
```

Comment:
I think it's better to have the highlighted city as a static value of `env_t` rather than `dynamic_cast<city_info_t*>(win_get_top())` because the selected city is no longer highlighted once the city info window becomes not the top while still visible. Is it a deserved behavior?

Reply:
The question then becomes what if you have multiple city info dialogs open at the same time.
I guess that is fine when it just "unselects" all other cities when a different city is selected.